mod-source-record-manager-server/src/main/java/org/folio/verticle/DeleteJobExecutionPeriodicallyVerticle.java

```diff
@@ -0,0 +1,63 @@
+package org.folio.verticle;
+
+import io.vertx.core.AbstractVerticle;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dao.JobExecutionDao;
+import org.folio.dao.util.TenantUtil;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Log4j2
+@Component
+@PropertySource("classpath:application.properties")
+public class DeleteJobExecutionPeriodicallyVerticle extends AbstractVerticle {
+
+  private static AbstractApplicationContext springGlobalContext;
+
+  @Autowired
+  private JobExecutionDao jobExecutionDao;
+
+  @Value("${job.execution.permanent.delete.interval.ms:86400000}")
+  private long jobExecutionPermanentDeletionInterval;
+  private long timerId;
+
+  public static void setSpringContext(AbstractApplicationContext springContext) {
```

Comment:
We already have in SRM 2 verticles that doing some periodical jobs and it will be the third verticle. We can create new abstract class with name something like this AbstractPeriodicallyJobsVerticle and move all this spring stuff to abstract class. It will be easier to refactor in future when we change approach of initiatialization of spring context.
And we will store common things connected with periodically execution in 1 abstract class.

Reply:
I made this refactoring, so resolving the comment